assets/js/atomic/utils/blocks-registration-manager/block-registration-strategy.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * External dependencies
+ */
+import {
+	BlockConfiguration,
+	registerBlockType,
+	unregisterBlockType,
+	registerBlockVariation,
+	unregisterBlockVariation,
+	BlockVariation,
+	BlockAttributes,
+} from '@wordpress/blocks';
+
+export interface BlockRegistrationStrategy {
+	register(
+		blockName: string | Partial< BlockConfiguration >,
+		blockSettings: Partial< BlockConfiguration >
+	): boolean;
+	unregister( blockName: string, variationName?: string ): boolean;
+}
+
+export class BlockTypeStrategy implements BlockRegistrationStrategy {
+	register(
+		blockName: string,
+		blockSettings: Partial< BlockConfiguration >
+	): boolean {
+		return Boolean( registerBlockType( blockName, blockSettings ) );
```

Comment:
Yes, unfortunately the current version that we are using for @woocommerce/blocks is not correctly typed. It accepts both [`blockName` ](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-registration/#block-name)or the [`blockConfiguration`](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-registration/#block-configuration); however, it errors out when using the first one. 

I added the following code to prevent the error from appearing. When we update the package to a version that contains the correct types, an error will occur so we know that we can fix this and remove the comment (this approach is being used in other places in our codebase):
```
// @ts-expect-error: `registerBlockType` is typed in WordPress core
```

Reply:
Thanks for the explanation. It makes sense! :+1: